modules/frontend/metrics_query_range_sharder.go

```diff
@@ -319,12 +438,14 @@ func (s *queryRangeSharder) generatorRequest(searchReq tempopb.QueryRangeRequest
 
 	searchReq.QueryMode = querier.QueryModeRecent
 
-	// No sharding on the generators (unnecessary), but we do apply sampling
-	// rates.  In this case we execute a single arbitrary shard. Choosing
-	// the last shard works. The first shard should be avoided because it is
-	// weighted slightly off due to int63/128 sharding boundaries.
-	searchReq.ShardID = uint32(1.0 / samplingRate)
-	searchReq.ShardCount = uint32(1.0 / samplingRate)
+	if s.replicationFactor == backend.DefaultReplicationFactor {
```

Comment:
I'm ok with deleting this entirely.  Otherwise the check could be clearer against s.cfg.QueryGeneratorBlocks?